netdump.c

```diff
@@ -301,6 +318,77 @@ int get_broadcast_inet(int sockfd, char *iface, struct sockaddr_in *brcast)
     return 1;
 }
 
+/*______________________________________________*\
+ *     Check if interface is loopback *
+\*______________________________________________*/
+int is_loopback_interface(int sockfd, char *iface)
+{
+    struct ifreq ifr;
+
+    strncpy(ifr.ifr_name, iface, IFNAMSIZ - 1);
+    if (ioctl(sockfd, SIOCGIFFLAGS, &ifr) < 0)
+        return FALSE;
+    
+    return (ifr.ifr_flags & IFF_LOOPBACK) ? TRUE : FALSE;
+}
+
+/*______________________________________________*\
+ * Simple hash function for packet deduplication *
+\*______________________________________________*/
+uint32_t packet_hash(const packet_t * packet)
```

Comment:
That logic is quite bad and will not work in many cases that are very important to show in a network packet inspecter. Such as consecutive ICMP packet generated by ping, they will have mostly same length and be very similar (ECHO packets)

I would suggest you implement a hash not based on the packet_t but something like a JENKINS hash that can be done on the full ethernet packet received by recv

It will be slow, but it only impact loopback usecase, so I accept that cost.